frontend/apps/filemanager/filemanagerutil.lua

```diff
@@ -138,34 +130,64 @@ end
         text = _("Reset"),
         id = "reset", -- used by covermenu
-        enabled = (not button_disabled and DocSettings:hasSidecarFile(ffiutil.realpath(file))) and true or false,
+        enabled = (not button_disabled and (sidecar_file or custom_metadata_file or custom_cover_file)) and true or false,
         callback = function()
+            local CheckButton = require("ui/widget/checkbutton")
             local ConfirmBox = require("ui/widget/confirmbox")
+            local check_button_settings, check_button_cover, check_button_metadata
             local confirmbox = ConfirmBox:new{
                 text = T(_("Reset this document?") .. "\n\n%1\n\n" ..
-                    _("Document progress, settings, bookmarks, highlights, notes and custom cover image will be permanently lost."),
+                         _("Resetted information will be permanently lost."),
                     BD.filepath(file)),
                 ok_text = _("Reset"),
                 ok_callback = function()
-                    local custom_metadata_purged = filemanagerutil.purgeSettings(file)
-                    if custom_metadata_purged then -- refresh coverbrowser cached book info
-                        local FileManager = require("apps/filemanager/filemanager")
-                        local ui = FileManager.instance
-                        if not ui then
-                            local ReaderUI = require("apps/reader/readerui")
-                            ui = ReaderUI.instance
-                        end
+                    local data_to_purge = {
+                        doc_settings         = check_button_settings.checked,
+                        custom_cover_file    = check_button_cover.checked and custom_cover_file,
+                        custom_metadata_file = check_button_metadata.checked and custom_metadata_file,
+                    }
+                    DocSettings:open(file):purge(nil, data_to_purge)
+                    if data_to_purge.custom_cover_file or data_to_purge.custom_metadata_file then
+                        local ui = require("apps/filemanager/filemanager").instance
+                                or require("apps/reader/readerui").instance
                         if ui and ui.coverbrowser then
-                            ui.coverbrowser:deleteBookInfo(file)
+                            ui.coverbrowser:deleteBookInfo(file) -- refresh coverbrowser cached book info
                         end
                     end
-                    require("readhistory"):fileSettingsPurged(file)
+                    if data_to_purge.doc_settings then
+                        require("readhistory"):fileSettingsPurged(file)
+                    end
                     caller_callback()
                 end,
             }
+            check_button_settings = CheckButton:new{
+                text = _("document settings, progress, bookmarks, highlights, notes"),
+                checked = sidecar_file and true or false,
```

Comment:
But sidecar_file is already `true` or `false`? :-)

Reply:
https://github.com/koreader/koreader/blob/ed2ea6803f4b9be6b065f43590229dea4fe51426/frontend/docsettings.lua#L93-L97

Reply:
Er, that function name should be changed go *get* ASAP if that's the case imho.

Reply:
But anyway, at the very least I'd do `has_sidecar_file = hasSidecarFile() and true or false` rather than doing it over and over in the results so that the workaround is contained.

Reply:
IIRC, there's precedent for this because of that very inconsistency in name vs. function (which has come up multiple times before, so, for once, I'd bite the bullet and *fix all the things* there ;p).

Reply:
Depending on the current callers, yeah, a wrapper is not a bad idea (and tail calls are basically free in LuaJIT, so it's pretty harmless).

Reply:
> IIRC, there's precedent for this because of that very inconsistency in name vs. function (which has come up multiple times before, so, for once, I'd bite the bullet and fix all the things the right way there ;p).

The pattern under discussion is completely unique to this PR (and should not exist).

```
$ rg "has[A-Z][a-z]*\(\) and"
frontend/dispatcher.lua
80:    toggle_key_repeat = {category="none", event="ToggleKeyRepeat", title=_("Toggle key repeat"), device=true, condition=Device:hasKeys() and Device:canKeyRepeat(), separator=true},

frontend/device/generic/device.lua
149:    if self:hasKeys() and self.input and self.input.event_map then

plugins/gestures.koplugin/defaults.lua
9:        tap_left_bottom_corner = Device:hasFrontlight() and {toggle_frontlight = true,} or nil,
14:        one_finger_swipe_left_edge_down = Device:hasFrontlight() and {decrease_frontlight = 0,} or nil,
15:        one_finger_swipe_left_edge_up = Device:hasFrontlight() and {increase_frontlight = 0,} or nil,
61:        two_finger_swipe_south = Device:hasFrontlight() and {decrease_frontlight = 0,} or nil,
62:        two_finger_swipe_north = Device:hasFrontlight() and {increase_frontlight = 0,} or nil,
76:        tap_left_bottom_corner = Device:hasFrontlight() and {toggle_frontlight = true,} or nil,
81:        one_finger_swipe_left_edge_down = Device:hasFrontlight() and {decrease_frontlight = 0,} or nil,
82:        one_finger_swipe_left_edge_up = Device:hasFrontlight() and {increase_frontlight = 0,} or nil,
128:        two_finger_swipe_south = Device:hasFrontlight() and {decrease_frontlight = 0,} or nil,
129:        two_finger_swipe_north = Device:hasFrontlight() and {increase_frontlight = 0,} or nil,
```